olp-cpp-sdk-dataservice-read/src/StreamLayerClientImpl.cpp

```diff
@@ -155,6 +155,82 @@ client::CancellableFuture<SubscribeResponse> StreamLayerClientImpl::Subscribe(
       std::move(cancel_token), std::move(promise));
 }
 
+client::CancellationToken StreamLayerClientImpl::Unsubscribe(
+    UnsubscribeResponseCallback callback) {
+  auto unsubscribe_task =
+      [=](client::CancellationContext context) -> UnsubscribeResponse {
+    std::string subscription_id;
+    std::string subscription_mode;
+    std::string x_correlation_id;
+
+    {
+      std::lock_guard<std::mutex> lock(mutex_);
+      if (!client_context_) {
+        OLP_SDK_LOG_WARNING_F(kLogTag,
+                              "Unubscribe: unsuccessful, not subscribed");
+
+        return client::ApiError(client::ErrorCode::InvalidArgument,
+                                "Not subscribed", false);
+      }
+
+      subscription_id = client_context_->subscription_id;
+      subscription_mode = client_context_->subscription_mode;
+      x_correlation_id = client_context_->x_correlation_id;
+    }
+
+    OLP_SDK_LOG_INFO_F(kLogTag,
+                       "Unsubscribe: started, subscription_id=%s, "
+                       "subscription_mode=%s, x_correlation_id=%s",
+                       subscription_id.c_str(), subscription_mode.c_str(),
+                       x_correlation_id.c_str());
+
+    auto stream_api = ApiClientLookup::LookupApi(
```

Comment:
Excerpt from the Platform documentation:

Deletes a subscription to a layer (topic). This operation removes the subscription from the service. The base path to use is the value of 'nodeBaseURL' returned from /subscribe POST request.